src/components/ReadingProgressPage/ReadingHistory/AddReading/index.tsx

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable max-lines */
+import { useMemo, useState, useContext, useEffect } from 'react';
+
+import useTranslation from 'next-translate/useTranslation';
+import { useSWRConfig } from 'swr';
+import useSWRImmutable from 'swr/immutable';
+
+import styles from './AddReading.module.scss';
+
+import VerseRangesList from '@/components/ReadingGoal/ReadingGoalAmount/VerseRangesList';
+import VerseRangeInput from '@/components/ReadingGoal/ReadingGoalInput/VerseRangeInput';
+import { isValidVerseRange } from '@/components/ReadingGoalPage/utils/validator';
+import DataContext from '@/contexts/DataContext';
+import Button, { ButtonShape, ButtonSize, ButtonVariant } from '@/dls/Button/Button';
+import Calendar from '@/dls/Calendar';
+import DurationInput from '@/dls/DurationInput';
+import { ModalSize } from '@/dls/Modal/Content';
+import Modal from '@/dls/Modal/Modal';
+import Spinner from '@/dls/Spinner/Spinner';
+import { ToastStatus, useToast } from '@/dls/Toast/Toast';
+import useGetMushaf from '@/hooks/useGetMushaf';
+import ChevronLeft from '@/icons/chevron-left.svg';
+import ChevronRight from '@/icons/chevron-right.svg';
+import ArrowLeft from '@/icons/west.svg';
+import { ActivityDayType } from '@/types/auth/ActivityDay';
+import { getFilterActivityDaysParamsOfCurrentMonth } from '@/utils/activity-day';
+import { estimateReadingTime, updateActivityDay } from '@/utils/auth/api';
+import {
+  makeEstimateReadingTimeUrl,
+  makeFilterActivityDaysUrl,
+  makeStreakUrl,
+} from '@/utils/auth/apiPaths';
+import {
+  dateToDateString,
+  dateToReadableFormat,
+  dateToYearMonthDay,
+  getCurrentMonth,
+  getMonthDateObject,
+  getMonthsInYear,
+} from '@/utils/datetime';
+import { logValueChange, logButtonClick, logFormSubmission } from '@/utils/eventLogger';
+
+const AddReading = () => {
+  const chaptersData = useContext(DataContext);
+  const { t, lang } = useTranslation('reading-progress');
+  const [isOpen, setIsOpen] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [rangeStart, setRangeStart] = useState<string | null>(null);
+  const [rangeEnd, setRangeEnd] = useState<string | null>(null);
+  const [ranges, setRanges] = useState<string[]>([]);
+  const [selectedMonth, setSelectedMonth] = useState<number>(() => getCurrentMonth());
+  const [selectedDate, setSelectedDate] = useState<string | null>(null);
+  const selectedYear = useMemo(() => new Date().getFullYear(), []);
+  const months = useMemo(() => getMonthsInYear(selectedYear, lang), [selectedYear, lang]);
+  const mushaf = useGetMushaf();
+  const toast = useToast();
+  const { cache, mutate } = useSWRConfig();
+
+  const [totalSeconds, setTotalSeconds] = useState<number>(0);
+  const { isValidating, data } = useSWRImmutable(makeEstimateReadingTimeUrl({ ranges }), () =>
```

Comment:
This get the API called just by visiting /reading-goal/progress even without having opened the Add reading modal. Having said that, I believe we should lazy load the modal until the add reading button is clicked
![Sep-03-2023 11-23-42](https://github.com/quran/quran.com-frontend-next/assets/15169499/aadfe4ea-a4f4-43ac-a6a4-fff599efb56e)